### PR TITLE
Improved regex validation of version in GH repository name.

### LINKED ISSUE #646: Image create from repository ends up with error, when version is specified in argument
**Describe the bug**
Specifying version in project name do not work, when creating image from GH project

**To Reproduce**
1. Run: `./pharo-launcher image create fromRepo --newImageName RoassalPlayground pharo-graphics/Roassal:v1.05b`
2. See error: `PhLError: Invalid repository name: pharo-graphics/Roassal:v1.05b. Please specify in {owner}/{project} or {owner}/{project}:{feature}/{branch-name} format`

**Expected behavior**
Project is loaded successfully into new image.

**Version information:**
 - OS: Linux/Ubuntu
 - Version: https://github.com/pharo-project/pharo-launcher/actions/runs/7501196046

**Expected development cost**
Fix regex checking repo name to consume version tag as well. See: `PhLImageCreateFromRemoteRepoCliCommand >> regexForRepoName`

**Additional context**
Reported by user on Discord: https://discord.com/channels/223421264751099906/1191849727046734024/1195796060321353728

### PR DESCRIPTION
**Pull request by Bajger**
- fixes #646

### CODE REVIEW

**Review by demarey — COMMENTED** [review]
I do not understand why you need to validate the version with a regex. It could ba anything: a branch name, a version like v1.2 or just 1.2 or 1.0-RC-somecomment, etc.
For me, it is not parsable. Do you really nead it?
From my understanding,  you just need to ensure owner/project:branch-or-version.
Am I right?

**Comment by Bajger** [comment]
Right, maybe it is too strict. But how you'd envision validation of: `branch-or-version` ? It cound't contain all special characters, right? Only `/`, `-`, `.`, anything else?

**Comment by Bajger** [comment]
@demarey Should be more universal now: version or branch includes letters, numbers or `-`, `/`, `.`, `_`

File: src/PharoLauncher-CLI-Tests/PhLImageCreateFromRemoteRepoCliCommandTest.class.st [hunk 1]
```diff
@@ -78,7 +78,7 @@ PhLImageCreateFromRemoteRepoCliCommandTest >> testValidateRepoFullName [
 	| commandSpec validRepoNames |
 	
 	commandSpec := PhLImageCreateFromRemoteRepoCliCommand asCliCommand.
-	validRepoNames := #('owner/project' 'owner/project:branch' 'owner/some-project:feature/branch-name').
+	validRepoNames := #('owner/project' 'owner/project:branch' 'owner/some-project:feature/branch-name' 'owner/project:v1' 'owner/project:v1.1' 'owner/project:v1.1a' 'owner/project:V1.1.1a' 'owner/project:1.0-RC-somecomment').
```
Comment (demarey) [inline src/PharoLauncher-CLI-Tests/PhLImageCreateFromRemoteRepoCliCommandTest.class.st:81]: I would rewrite the test with asserts:
```smalltalk
testValidateRepoFullName

	| commandSpec validRepoNames invalidRepoName createCommand |
	
	commandSpec := PhLImageCreateFromRemoteRepoCliCommand asCliCommand.
	validRepoNames := #('owner/project' 'owner/project:branch' 'owner/some-project:feature/branch-name' 'owner/project:v1' 'owner/project:v1.1' 'owner/project:v1.1a' 'owner/project:V1.1.1a' 'owner/project:1.0-RC-somecomment').

	validRepoNames do: [:repoName | 
		createCommand := (commandSpec activationWith: { 'fromRepo'. repoName }) command.
		self 
			assert: createCommand isValidRepoFullName 
			description: 'Valid repository name not validated: ', repoName.
	].

	invalidRepoName := 'owner/project:'.
	createCommand := (commandSpec activationWith: { 'fromRepo'. invalidRepoName }) command.
	self 
		deny: createCommand isValidRepoFullName 
		description: 'Invalid repository name validated: ', invalidRepoName.
```

**Pull request merged**